core/services/request/CurrentPage.php

```diff
@@ -247,6 +247,9 @@ private function getPostType(WP $WP = null)
      */
     public function isEspressoPage()
     {
+        if ($this->is_espresso_page === null) {
+            $this->setEspressoPage();
+        }
```

Comment:
changed default value for property from `false` and then use `null` value to trigger check